src/main/java/pl/edu/agh/iet/katabank/InMemoryBankProductsRepository.java

```diff
@@ -0,0 +1,49 @@
+package pl.edu.agh.iet.katabank;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class InMemoryBankProductsRepository implements BankProductsRepository {
+
+    private Map<Customer, Set<Account>> accountsMap;
+    private Map<Account, Set<Deposit>> depositsMap;
+
+    public InMemoryBankProductsRepository() {
+        this.accountsMap = new HashMap<>();
+        this.depositsMap = new HashMap<>();
+    }
+
+    @Override
+    public void addAccount (Account account){
+        Set<Account> customerAccounts = accountsMap.get(account.getOwner()) != null
+                ? accountsMap.get(account.getOwner()) : new HashSet<>();
+        customerAccounts.add(account);
+        accountsMap.put(account.getOwner(), customerAccounts);
+    }
+
+    @Override
+    public Set<Account> findAccountsForCustomer(Customer customer) {
+        return accountsMap.getOrDefault(customer, new HashSet<>());
+    }
+
+    @Override
+    public void addDeposit(Deposit deposit) {
+        Set<Deposit> accountsDeposits = depositsMap.get(deposit.getConnectedAccount()) != null
+                ? depositsMap.get(deposit.getConnectedAccount()) : new HashSet<>();
+        accountsDeposits.add(deposit);
+        depositsMap.put(deposit.getConnectedAccount(), accountsDeposits);
+    }
+
+    @Override
+    public Set<Deposit> findDepositsForCustomer(Customer customer) {
+        Set<Deposit> customersDeposits = new HashSet<>();
+        for (Account account : findAccountsForCustomer(customer)){
+            customersDeposits.addAll(depositsMap.getOrDefault(account, new HashSet<>()));
+        }
+        return customersDeposits;
```

Comment:
Write unit test(s) to check if all customer deposits connected with different accounts of this customer are returned.
You could also test if deposit of another customer is not returned.

Reply:
Added tests in inMemoryBankProductsRepositoryTest